app/ui.js

```diff
@@ -965,9 +983,171 @@ var UI;
+            UI._ast2100_serverVideoSettings = undefined;
+        },
+        
+        // Updates the UI to reflect values received from the server.
+        ast2100_updateVideoSettings: function (videoSettings) {
+            Util.Info("AST2100 video settings changed:");
+            Util.Info(videoSettings);
+
+            // We use this to tell if the user changed anything when they apply
+            // settings.
+            UI._ast2100_serverVideoSettings = videoSettings;
+            
+            // First run: tell UI to show video quality controls, now that we
+            // know we are on a machine that supports them, and we know their
+            // current values.
+            if (!UI._ast2100_videoSettingsInitialized) {
+                UI.slider("noVNC_setting_ast2100_quality", {minVal: 0x0, maxVal: 0xB});
+                this.ast2100_setSettingsVisible(true);
+                UI._ast2100_videoSettingsInitialized = true;
+            }
+            
+            // Average the two quant table selectors as a poor way of dealing
+            // with the fact that they can, technically, be different.
+            var quality = ~~((videoSettings.quantTableSelectorLuma + videoSettings.quantTableSelectorChroma) / 2);
+            document.getElementById("noVNC_setting_ast2100_quality").setValue(quality);
+            
+            // Either 444 or 422 (which is really 4:2:0).
+            document.getElementById("noVNC_setting_ast2100_subsampling").value = videoSettings.subsamplingMode;
+        },
+
+        // Returns the current state of the UI.
+        ast2100_getConfiguredSettings: function () {
+            var quality = document.getElementById("noVNC_setting_ast2100_quality").value;
+            return {
+                quantTableSelectorLuma: quality,
+                quantTableSelectorChroma: quality,
+                subsamplingMode: parseInt(document.getElementById("noVNC_setting_ast2100_subsampling").value)
+            };
+        },
+
+        ast2100_setSettingsVisible: function (visible) {
+            var propertyValue;
+            if (visible) {
+                propertyValue = 'block';
+                Util.Info('Showing AST2100 UI.');
+            } else {
+                Util.Info('Hiding AST2100 UI.');
+                propertyValue = 'none';
+            }
+            
+            document.getElementById("noVNC_ast2100_settings").style.display = propertyValue;
+        },
+
+        // Create a slider widget.
+        slider: function(id, defaults) {
+            var slider = document.getElementById(id),
+                btn = slider.children[0],
+                btnWidth = 8,
+                sldWidth = 120,
+                down = false,
+                minVal = defaults.minVal,
+                maxVal = defaults.maxVal,
+                defaultValue = defaults.value,
+                snapToInt = true,
+                snapDuringDrag = true,
+                rect;
```

Comment:
Why not a standard HTML element? A lot of work in order to construct your own widget.